src/lib/config.ts

```diff
@@ -1522,6 +1517,24 @@ export async function update() {
                 "vmaps",
             ].forEach(settingName => updateAll([settingName], updateSetting))
             set("configversion", "1.9")
+        }
+        case "1.9": {
+            const local = (await browser.storage.local.get(CONFIGNAME))[CONFIGNAME] as {storageloc?: "local" | "sync"}
+            const sync = (await browser.storage.sync.get(CONFIGNAME))[CONFIGNAME] as {storageloc?: "local" | "sync"}
+            // Possible combinations:
+            // storage:storageloc_setting => winning storageloc setting
+            // l:l, s:* => l
+            // l:undefined, s:l =>  l
+            // l:undefined, s:s => s
+            // l: undefined, s:undefined => s
+            // l:s, s:* =>  s
+            const current_storageloc = local?.storageloc !== undefined ? local.storageloc : sync?.storageloc !== undefined ? sync.storageloc : "sync"
```

Comment:
Just a style preference: I think nested ternary operations are unreadable and need parens. Since the line is long breaking it apart might help too (e.g. `X\n? Y\n: Z`). Feel free to ignore me if you really prefer the current way it's written :)